src/marshmallow/schema.py

```diff
@@ -497,7 +498,7 @@ def _call_and_store(getter_func, data, *, field_name, error_store, index=None):
             return error.valid_data or missing
         return value
 
-    def _serialize(self, obj: _T | typing.Iterable[_T], *, many: bool = False):
+    def _serialize(self, obj: dict | typing.Iterable[dict], *, many: bool = False):
```

Comment:
this isn't quite right, is it? `obj` can be any arbitrary object type or an iterable objects.

Reply:
Per https://github.com/marshmallow-code/marshmallow/pull/2181/files#r1401344892, the issue was that `_T` was being used incorrectly in `_deserialize`. Moving it to `load()` would avoid the need to change this `TypeVar`.

Reply:
Per https://github.com/marshmallow-code/marshmallow/pull/2181#discussion_r1406499380 `obj` should be `Any`.